### PR TITLE
Rivest

### PR DESCRIPTION
**Pull request by silasdavis**

### CODE REVIEW

File: core/app.go [hunk 1]
```diff
@@ -77,8 +77,7 @@ func (a *App) validateChainConditions() error {
 			return fmt.Errorf("chain with id %s does not exist", chain.ChainId)
 		}
 
-		// check sender is the valid address
-		addr, err := sdk.GetFromBech32(chain.Sender, chain.AccountPrefix)
+		ki, err := client.Keybase.Key(chain.KeyName())
```
Comment (amaury1093) [inline core/app.go:80]: Why is this change needed? Would it have been simpler to keep the `key_name/sender` structure and overwrite them with correct values on infraco?

File: core/server.go [hunk 2]
```diff
@@ -80,16 +81,21 @@ func (s *Server) GiveMe(ctx context.Context, request *faucetpb.GiveMeRequest) (*
...
 
-	// validate address format
-	acc, err := sdk.GetFromBech32(request.Address, chainConfig.AccountPrefix)
-	if err != nil {
-		return nil, status.Error(codes.InvalidArgument, "invalid address")
+	var address []byte
+	var err error
+	if common.IsHexAddress(request.Address) {
+		address = common.HexToAddress(request.Address).Bytes()
+	} else {
```
Comment (amaury1093) [inline core/server.go:91]: Propose to add some comments, it took an extra cycle too much of thinking to understand what you were doing here

```suggestion
	// Accept both Ethereum and Cosmos addresses
	if common.IsHexAddress(request.Address) {
		address = common.HexToAddress(request.Address).Bytes()
	} else {
```

File: Makefile [hunk 3]
```diff
@@ -1,20 +1,32 @@
...
-.PHONEY: lint
+.PHONY: lint
 lint:
 	docker run -v $$(pwd):/src -w /src --rm bufbuild/buf:$(BUF_VERSION) lint
-	docker run -v $$(pwd):/src -w /src --rm bufbuild/buf:$(BUF_VERSION) breaking --against 'https://github.com/johanbrandhorst/grpc-gateway-boilerplate.git#branch=master'
\ No newline at end of file
+	docker run -v $$(pwd):/src -w /src --rm bufbuild/buf:$(BUF_VERSION) breaking --against 'https://github.com/johanbrandhorst/grpc-gateway-boilerplate.git#branch=master'
+
+
+DOCKER_VERSION:=v1.0.2
```
Comment (amaury1093) [inline Makefile:27]:
```suggestion
COSMFAUCET_VERSION:=v1.0.2
```

**Review by amaury1093 — COMMENTED** [review]
Would like to understand the key_name/sender refactor before approving, otherwise LGTM.

File: core/app.go (hunk 1 above)
Comment (silasdavis) [reply core/app.go:80]: The thing that was weird with this approach is that the config was asking you for three things that are modelled as independent but are not/do not need to be:

1. The `Key` AKA mnemonic that uniquely defines the `Sender` address, you basically need to derive that yourself from the key in the current version
2. The `KeyName` which throws an error if you attempt to `RestoreKey` to that name
3. The `Sender` address which was as above

The point here is we remove 3 bits of config, and replace them with 1 piece of config and derive everything from that.

> Would it have been simpler to keep the key_name/sender structure and overwrite them with correct values on infraco?

So not really, it would have been a load of busywork to go and compute these things first when the faucet itself could do that.

File: core/server.go (hunk 2 above)
Comment (silasdavis) [reply core/server.go:91]: Yep, agreed, will add.

**Review by amaury1093 — APPROVED** [review]
